### PR TITLE
feat(Timetable): Include track assignment

### PR DESCRIPTION
**Pull request by KaylaBrady**
#### Summary of changes

**Asana Ticket:** [TICKET_NAME](TICKET_LINK)

---

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

#### New endpoints, or non-trivial changes to current endpoints
* [ ] **Have we load-tested** any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [ ] **If this change involves routes,** does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?

### CODE REVIEW

File: apps/site/lib/site_web/templates/schedule/_timetable_schedule.html.eex [hunk 1]
```diff
@@ -3,18 +3,30 @@
...
-          <%= content_tag(:span, "Early Departure", class: "sr-only") %>
+    <span class="m-timetable__left-icons">
+      <span class="m-timetable__track-change-icon">
+        <%= if @track_change != nil do %>
+
+          <%= content_tag(:i, "", class: "track-change-icon notranslate fa fa-shuffle no-margin-right", aria: [hidden: "true"]) %>
+            <%= content_tag(:span, @track_change, class: "sr-only") %>
+
+          <% else %>
+            &nbsp;
```
Comment (kotva006) [inline apps/site/lib/site_web/templates/schedule/_timetable_schedule.html.eex:14]: nit-pick: The spacing looks a little weird here

File: apps/site/lib/site_web/controllers/schedule/timetable_controller.ex [hunk 2]
```diff
@@ -58,16 +58,63 @@ defmodule SiteWeb.ScheduleController.TimetableController do
       all_stops: all_stops
     } = build_timetable(conn.assigns.all_stops, timetable_schedules)
 
+    canonical_rps =
+      route.id
+      |> RoutePatterns.Repo.by_route_id(direction_id: direction_id, canonical: true)
```
Comment (kotva006) [inline apps/site/lib/site_web/controllers/schedule/timetable_controller.ex:63]: nit-pick: I thought the linter usually complains about piping into a single function

File: apps/site/lib/site_web/controllers/schedule/timetable_controller.ex [hunk 3]
```diff
@@ -58,16 +58,63 @@ defmodule SiteWeb.ScheduleController.TimetableController do
...
+
+    # Don't use the stop ids set on the route pattern - those are mapped to the parent stops.
+    # Get the stops directly for the canonical trips
+    canonical_stops =
```
Comment (kotva006) [inline apps/site/lib/site_web/controllers/schedule/timetable_controller.ex:67]: nit-pick: rename this clarify it holds the stop ids and not the stops

File: apps/site/lib/site_web/templates/schedule/_timetable.html.eex [hunk 4]
```diff
@@ -125,7 +126,7 @@
                   <div class="sr-only"><%= full_trip_message %></div>
                   <div aria-hidden="true"><%= trip_message %></div>
                 <% else %>
-                  <%= render "_timetable_schedule.html", Map.merge(assigns, %{trip_schedule: trip_schedule, schedule: schedule, stop: stop}) %>
+                  <%= render "_timetable_schedule.html", Map.merge(assigns, %{trip_schedule: trip_schedule, schedule: schedule, stop: stop, track_change: track_change}) %>
```
Comment (kotva006) [inline apps/site/lib/site_web/templates/schedule/_timetable.html.eex:129]: nit-pick: Reformat this so its not that long of a line

File: apps/stops/test/repo_test.exs [hunk 5]
```diff
@@ -1,4 +1,7 @@
 defmodule Stops.RepoTest do
+  @moduledoc """
+  TODO: Mock API data in this whole file
```
[the remark this answers is not in the document]
Comment (KaylaBrady) [reply apps/stops/test/repo_test.exs:3]: Credit goes to @thecristen for adding this comment!

File: apps/site/lib/site_web/controllers/schedule/timetable_controller.ex (hunk 2 above)
Comment (KaylaBrady) [reply apps/site/lib/site_web/controllers/schedule/timetable_controller.ex:63]: Looks like that dotcom hasn't opted into that particular check (`Credo.Check.Readability.SinglePipe`  configured with `false` [here](https://github.com/mbta/dotcom/blob/5f720af3324ac8b60aaca29f5f077195270f99d6/config/.credo.exs#L150). But I'm a fan of that check & will update here!

Comment (kotva006) [reply apps/site/lib/site_web/controllers/schedule/timetable_controller.ex:63]: Okay, probably getting confused with Pick Assist configuration

**Review by i0sea — APPROVED** [review]
looks good to me

**Pull request merged**